src/lib/rng/stateful_rng/stateful_rng.cpp

```diff
@@ -34,68 +34,47 @@ bool Stateful_RNG::is_seeded() const
    return m_reseed_counter > 0;
    }
 
-void Stateful_RNG::add_entropy(const uint8_t input[], size_t input_len)
-   {
-   lock_guard_type<recursive_mutex_type> lock(m_mutex);
-
-   update(input, input_len);
-
-   if(8*input_len >= security_level())
-      {
-      reset_reseed_counter();
-      }
-   }
-
-void Stateful_RNG::initialize_with(const uint8_t input[], size_t len)
+void Stateful_RNG::initialize_with(std::span<const uint8_t> input)
    {
    lock_guard_type<recursive_mutex_type> lock(m_mutex);
 
    clear();
-   add_entropy(input, len);
-   }
-
-void Stateful_RNG::randomize(uint8_t output[], size_t output_len)
-   {
-   randomize_with_input(output, output_len, nullptr, 0);
+   add_entropy(input);
    }
 
-void Stateful_RNG::randomize_with_ts_input(uint8_t output[], size_t output_len)
```

Comment:
Do you know why for the `Stateful_RNG:` the used input for `randomize_with_ts_input()` was a little different than for the general implementation?

I think it is good to unify what input is used, just want to make sure the change is intentional and OK to do.

Reply:
My understanding is that `Stateful_RNG` wants to be a bit more conservative to make sure that the state is always reasonably seeded.